doc/modules/partial_dependence.rst

```diff
@@ -238,7 +238,7 @@ if a split node involves an input feature of interest, the corresponding left
 or right branch is followed; otherwise both branches are followed, each branch
 being weighted by the fraction of training samples that entered that branch.
 Finally, the partial dependence is given by a weighted average of all the
-visited leaves values.
+visited leaves' values.
```

Comment:
Honestly I am not sure this pickiness is needed but I don't think it's worth spending too much time arguing about it either.

"the leaves value" vs "the leaves' value".
below: "the model's statistical performance" vs "the model statistical performance" 

Out of curiosity, I'd be interested to have a native speaker take on it e.g. @betatim  or @thomasjpfan.

I always assumed the 's was not more correct but not really needed but maybe my assumption comes from the fact that it's not as obvious in spoken language.

I read a bit about it e.g. https://dictionary.cambridge.org/grammar/british-grammar/possession-john-s-car-a-friend-of-mine and they don't mention that the 's is optional.

Reply:
The `'s` or `s'` is not optional here. For a PR focused on improving documentation, I think it's okay to also improve the grammar.